### PR TITLE
tools: add symbiflow support and re-enable CI on all supported tools

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

This PR adds symbiflow support and re-enables all exhaustive tests.

### CODE REVIEW

**Comment by acomodi** [comment]
@mithro Seems all tests passed, I may introduce here nextpnr as well if https://github.com/SymbiFlow/conda-packages/pull/89 gets merged:

```
Merging results
+-------------------------+--------------+---------+--------------+--------+---------------------+--------+
| Project                 | Toolchain    | Family  | Part         | Board  | Options             |        |
+-------------------------+--------------+---------+--------------+--------+---------------------+--------+
| ibex                    | vivado       | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| ibex                    | vpr          | xc7     | a35tcsg324-1 | arty   | pcf_sdc_xdc_carry-n | passed |
| ibex                    | yosys-vivado | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| litex-linux             | vivado       | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| litex-linux             | vpr          | xc7     | a35tcsg324-1 | arty   | pcf_sdc_xdc_carry-n | passed |
| murax                   | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| murax                   | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_sdc_carry-n     | passed |
| oneblink                | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| oneblink                | vivado       | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| oneblink                | vivado       | xc7     | z010clg400-1 | zybo   | xdc_carry-n         | passed |
| oneblink                | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_carry-n         | passed |
| oneblink                | vpr          | xc7     | a35tcsg324-1 | arty   | pcf_carry-n         | passed |
| oneblink                | vpr          | xc7     | z010clg400-1 | zybo   | pcf_carry-n         | passed |
| oneblink                | yosys-vivado | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| oneblink                | yosys-vivado | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| oneblink                | yosys-vivado | xc7     | z010clg400-1 | zybo   | xdc_carry-n         | passed |
| picorv32-wrap           | vivado       | xc7     | a35tcsg324-1 | arty   | xdc_carry-n         | passed |
| picorv32-wrap           | vivado       | xc7     | z010clg400-1 | zybo   | xdc_carry-n         | passed |
| picorv32-wrap           | vpr          | xc7     | a35tcsg324-1 | arty   | pcf_carry-n         | passed |
| picorv32-wrap           | vpr          | xc7     | z010clg400-1 | zybo   | pcf_carry-n         | passed |
| picosoc-simpleuart-wrap | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| picosoc-simpleuart-wrap | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_carry-n         | passed |
| picosoc-spimemio-wrap   | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| picosoc-spimemio-wrap   | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_carry-n         | passed |
| picosoc-wrap            | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| picosoc-wrap            | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_sdc_carry-n     | passed |
| vexriscv-verilog        | vivado       | xc7     | a35tcpg236-1 | basys3 | xdc_carry-n         | passed |
| vexriscv-verilog        | vpr          | xc7     | a35tcpg236-1 | basys3 | pcf_carry-n         | passed |
+-------------------------+--------------+---------+--------------+--------+---------------------+--------+
| Passed:                 | 28           | Failed: | 0            |        |                     | 100%   |
+-------------------------+--------------+---------+--------------+--------+---------------------+--------+
```

File: Makefile [hunk 1]
```diff
@@ -9,10 +9,13 @@ conda:
 	git submodule update --init --recursive
 	mkdir -p env
 	source utils/conda.sh
+	# FIXME: make this dynamic
```
Comment (mithro) [inline Makefile:12]: Please create an issue tracking this.

File: conf/requirements.txt [hunk 2]
```diff
@@ -8,3 +8,4 @@ python-constraint
 intervaltree
 jinja2
 pytest
+textx
```
Comment (mithro) [inline conf/requirements.txt:11]: Why is this needed? My guess is because the fasm library isn't being installed with `pip` correctly so it isn't seeing the dependencies?

File: utils/environment.python.sh [hunk 3]
```diff
@@ -2,9 +2,7 @@
 
 BASE_DIR=${FPGA_TOOL_PERF_BASE_DIR}
 # FIXME: fasm should be installed into the running Python environment.
-# FIXME: enable SYMBIFLOW when installed toolchain is available
-#export PYTHONPATH="${SYMBIFLOW}/python:$BASE_DIR:$BASE_DIR/third_party/fasm:$BASE_DIR/third_party/edalize:$PYTHONPATH"
-export PYTHONPATH="$BASE_DIR:$BASE_DIR/third_party/fasm:$BASE_DIR/third_party/edalize:$PYTHONPATH"
+export PYTHONPATH="${SYMBIFLOW}/bin/python:$BASE_DIR:$BASE_DIR/third_party/fasm:$BASE_DIR/third_party/edalize:$PYTHONPATH"
```
Comment (mithro) [inline utils/environment.python.sh:5]: This shouldn't be needed.

Use `pip install -e $BASE_DIR/third_party/fasm` or even better add it into the requirements.txt file

**Comment by litghost** [comment]
Should there be output files from the run, containing details like logs from the tools, final timing results, etc?

**Comment by acomodi** [comment]
@litghost Yeah, I am updating the CI config to upload the final results produced

**Comment by acomodi** [comment]
@litghost Done, all log files are now copied in the artifacts.

File: utils/environment.python.sh (hunk 3 above)
Comment (acomodi) [reply utils/environment.python.sh]: Done, added also edalize in the requirements

File: conf/requirements.txt (hunk 2 above)
Comment (acomodi) [reply conf/requirements.txt]: Yes, that was the issue, removed textx

File: Makefile (hunk 1 above)
Comment (acomodi) [reply Makefile]: Done

File: utils/environment.python.sh [hunk 4]
```diff
@@ -2,9 +2,7 @@
 
 BASE_DIR=${FPGA_TOOL_PERF_BASE_DIR}
 # FIXME: fasm should be installed into the running Python environment.
-# FIXME: enable SYMBIFLOW when installed toolchain is available
-#export PYTHONPATH="${SYMBIFLOW}/python:$BASE_DIR:$BASE_DIR/third_party/fasm:$BASE_DIR/third_party/edalize:$PYTHONPATH"
-export PYTHONPATH="$BASE_DIR:$BASE_DIR/third_party/fasm:$BASE_DIR/third_party/edalize:$PYTHONPATH"
+export PYTHONPATH="${SYMBIFLOW}/bin/python:$PYTHONPATH"
```
Comment (mithro) [inline utils/environment.python.sh:5]: This seems wrong and probably it doesn't needs to be set anymore?

Does this script do a `conda activate` somewhere?

Comment (acomodi) [reply utils/environment.python.sh]: Done, the script is actually not needed anymore as all PYTHON dependencies are already included.

**Comment by mithro** [comment]
@acomodi - Yay the @symbiflow-robot did the right thing I think? (Rerunning the CI and then merging when it went green?)

**Comment by mithro** [comment]
@acomodi Can you also log some github issues around cleaning up conda and python usage to match the `environment.yml` and `requirements.txt` style stuff so `PYTHONPATH` doesn't need to be changed in other repositories?

**Comment by acomodi** [comment]
> @acomodi - Yay the @symbiflow-robot did the right thing I think? (Rerunning the CI and then merging when it went green?)

Indeed. I am thinking though that this may be dangerous, in the sense that there should be at least one green review for each PR. If a `merge-if-green` label is erroneously added and no review has been done, the PR gets merged "unseen".

**Comment by acomodi** [comment]
> @acomodi Can you also log some github issues around cleaning up conda and python usage to match the `environment.yml` and `requirements.txt` style stuff so `PYTHONPATH` doesn't need to be changed in other repositories?

What other repositories you have in mind? I think this should be done only in repos that do use conda packages, which, as far as I know, there is arch-defs only (also sv-tests, but that makes use of environment.yml as well)

**Comment by mithro** [comment]
> > @acomodi - Yay the @symbiflow-robot did the right thing I think? (Rerunning the CI and then merging when it went green?)
> 
> Indeed. I am thinking though that this may be dangerous, in the sense that there should be at least one green review for each PR. If a `merge-if-green` label is erroneously added and no review has been done, the PR gets merged "unseen".

The code is at https://github.com/SymbiFlow/symbiflow-robot -- maybe log an issue?

**Comment by mithro** [comment]
> > @acomodi Can you also log some github issues around cleaning up conda and python usage to match the `environment.yml` and `requirements.txt` style stuff so `PYTHONPATH` doesn't need to be changed in other repositories?
> 
> What other repositories you have in mind? I think this should be done only in repos that do use conda packages, which, as far as I know, there is arch-defs only (also sv-tests, but that makes use of environment.yml as well)

Symbiflow-Arch-Defs is probably the big one. Project X-Ray uses conda too.
